### ISSUE TITLE
annex cannot handle filenames that start with a minus

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2131]
#### What is the problem?

```
% datalad add .
add(ok): /tmp/lit/-Pals.wcover.pdf (file)                                                                                    
...
Failed to run ['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'annex', 'add', '--json', u'-Pals.wcover.pdf', ..., '.'] under '/tmp/lit'. Exit code=1. out= err=Invalid option `-Pals.wcover.pdf'
```

Errors with annex help output displayed and exit code 1. Leave all files staged, nothing is commited. A subsequent `save` (no args) achieves the desired result.

Maybe we need to escape the file names that look like options when passing them to annex?

A subsequent `aggregate-metadata` command fails with an analog error of `annex metadata`.

Escaping the minus character seems to do the trick.

**Comment by yarikoptic** [comment]
We probably just need to add `--` before the list of the filenames

**Comment by yarikoptic** [comment]
Was fixed in #2134 which was merged into master in #2173
